### PR TITLE
Updated slicing operators in C-API to correct names.

### PR DESCRIPTION
**Pull request by hoytak**

### CODE REVIEW

File: src/capi/impl/capi_sarray.cpp [hunk 1]
```diff
@@ -723,11 +723,22 @@ EXPORT tc_sarray* tc_sarray_hash(const tc_sarray* sa, uint64_t salt, tc_error**
   ERROR_HANDLE_END(error, NULL);
 }
 
-EXPORT tc_sarray* tc_sarray_slice(const tc_sarray* sf, const int64_t start, const int64_t end, const int64_t stride, tc_error** error) {
+EXPORT tc_sarray* tc_sarray_slice(const tc_sarray* sf, const int64_t start, const int64_t slice, const int64_t end, tc_error** error) {
```
Comment (znation) [inline src/capi/impl/capi_sarray.cpp]: In the header, this parameter is named `stride` -- here it's named `slice`. Should we be consistent between the two?

File: src/capi/impl/capi_sarray.cpp [hunk 2]
```diff
@@ -723,11 +723,22 @@ EXPORT tc_sarray* tc_sarray_hash(const tc_sarray* sa, uint64_t salt, tc_error**
...
 
-  return new_tc_sarray(sf->value.subslice(start, end, stride));
+  return new_tc_sarray(
+      sf->value[{static_cast<long long>(start), static_cast<long long>(slice),
+                 static_cast<long long>(end)}]);
+
+  ERROR_HANDLE_END(error, NULL);
+}
+
+EXPORT tc_sarray* tc_sarray_subslice(const tc_sarray* sf, const int64_t start, const int64_t slice, const int64_t end, tc_error** error) {
```
Comment (znation) [inline src/capi/impl/capi_sarray.cpp]: Same comment as above.

File: src/capi/TuriCreate.h [hunk 3]
```diff
@@ -495,7 +495,8 @@ tc_sarray* tc_sarray_sort(const tc_sarray* sa, bool ascending, tc_error** error)
 bool tc_sarray_is_materialized(const tc_sarray* src, tc_error**);
 bool tc_sarray_size_is_known(const tc_sarray* src, tc_error**);
 tc_sarray* tc_sarray_hash(const tc_sarray* src, uint64_t salt, tc_error**);
-tc_sarray* tc_sarray_slice(const tc_sarray* sf, const int64_t start, const int64_t end, const int64_t stride, tc_error**);
+tc_sarray* tc_sarray_slice(const tc_sarray* sf, const int64_t start, const int64_t stride, const int64_t end, tc_error**);
+tc_sarray* tc_sarray_subslice(const tc_sarray* sf, const int64_t start, const int64_t stride, const int64_t end, tc_error**);
```
Comment (znation) [inline src/capi/TuriCreate.h]: Please add a unit test for the `subslice` function.

File: src/capi/impl/capi_sarray.cpp (hunk 2 above)
Comment (hoytak) [reply src/capi/impl/capi_sarray.cpp]: Oops, yes, cherry picked the wrong commit.

**Comment by hoytak** [comment]
This was actually addressed in another (approved) PR.  Closing for now.  Merge issues.

**Pull request closed without merging**